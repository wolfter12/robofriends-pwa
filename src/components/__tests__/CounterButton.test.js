import React from 'react';
import { shallow } from 'enzyme';

import CounterButton from '../CounterButton';

describe('CounterButton component', () => {
  it('renders CounterButton component', () => {
    const mockColor = 'red';

    const wrapper = shallow(<CounterButton color={mockColor} />);

    expect(wrapper.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('correctly increments the counter', () => {
    const mockColor = 'red';

    const wrapper = shallow(<CounterButton color={mockColor} />);

    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({ count: 2 });
  });
});
