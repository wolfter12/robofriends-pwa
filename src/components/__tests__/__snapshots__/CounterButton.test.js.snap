// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton component renders CounterButton component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  1
</button>
`;
